### PR TITLE
[stable/mongodb-replicaset] Add ServiceAccount option (#22694)

### LINKED ISSUE #22694: [stable/mongob-replicaset] Add ServiceAccount option
`ServiceAccount` support is missing in `mongodb-replicaset` chart.

**Is your feature request related to a problem? Please describe.**
I won't be able to use Consul Connect (with ACLs turned on) with MongoDB using `default` token. I have to create a ServiceAccount with specific name, which I can pass to Consul Connect annotations.

**Describe the solution you'd like**
I'd like to see an option like `serviceAccountName: <your-serviceaccount-name-here>` which will create ServiceAccount and pass it's name to Pod template StatefulSet.

**Describe alternatives you've considered**
I can just download the chart and patch it locally, but I believe that more people could benefit from this.

**Additional context**
N/A

### PR DESCRIPTION
**Pull request by krzysztof-miemiec**
#### Is this a new chart
Nope.

#### What this PR does / why we need it:
It adds a ServiceAccount option which is handy if you'd like to use this chart with Consul Connect.

#### Which issue this PR fixes
  - fixes #22694

#### Special notes for your reviewer:
N/A

#### Checklist
- [x] [DCO](https://github.com/helm/charts/blob/master/CONTRIBUTING.md#sign-your-work) signed
- [x] Chart Version bumped
- [x] Variables are documented in the README.md
- [x] Title of the PR starts with chart name (e.g. `[stable/mychartname]`)

### CODE REVIEW

File: stable/mongodb-replicaset/Chart.yaml [hunk 1]
```diff
@@ -1,7 +1,7 @@
 apiVersion: v1
 name: mongodb-replicaset
 home: https://github.com/mongodb/mongo
-version: 3.15.1
+version: 3.15.2
```
Comment (steven-sheehy) [inline stable/mongodb-replicaset/Chart.yaml]: Should be minor version bump for new features

File: stable/mongodb-replicaset/templates/mongodb-serviceaccount.yaml [hunk 2]
```diff
@@ -0,0 +1,6 @@
+{{- if .Values.serviceAccount -}}
+apiVersion: v1
+kind: ServiceAccount
+metadata:
```
Comment (steven-sheehy) [inline stable/mongodb-replicaset/templates/mongodb-serviceaccount.yaml:4]: Missing common labels and namespace that are present in other manifests.

**Review by steven-sheehy — CHANGES_REQUESTED** [review]
/assign
/ok-to-test

**Comment by krzysztof-miemiec** [comment]
Thanks for review @steven-sheehy! I added fixes in amend commit.

**Review by steven-sheehy — APPROVED** [review]
/lgtm

**Pull request merged**